docs/courses/csintro3/functions/comments.md

```diff
@@ -1,9 +1,213 @@
-# Comments for Functions
+# Activity: Function Documentation
 
-give description of standard function commenting
+In this section, ``||functions:functions||`` have been shown to have a wide variety of uses, with many different features.
 
+For those reasons, we will maintain a particular syntax for ``||functions:function||`` comments. This way, anyone who wants to learn about what the function do will have a consistent and easy way in which to identify these features.
+
+## Concept: Commenting Simple Functions
+
+The first functions discussed in the [Intro Lesson](/courses/csintro3/functions/intro) were fairly basic; they accepted no input, and did not produce any output.
+
+The ``||functions:sayHello||`` function is a simple demonstration of this; whenever it is called, it will ``||game:splash||`` "hello" to the person playing the game.
+
+To add a comment to a function like this, simply describe in a short sentence what the function is actually doing.
+
+## Example #1: Commenting on ``||functions:sayHello||``
+
+1. Review the code below
+2. Identify what the method will do
+3. Read the comment for the method: identify how it describes the method's behavior
+
+```typescript
+/**
+ * Displays a greeting for the player to read.
+ */
+function sayHello() {
+    game.splash("hello");
+}
+
+sayHello();
+```
+
+Notice that the comment does not go into much detail on the behavior - saying the exact greeting, or how it is done. If someone wants to know these things, they can read the code itself.
+
+Even if the function were changed to display different text in a slightly different way (as done below), the same comment could still be used to describe the function.
+
+```typescript
 /**
- * DESCRIPTION OF METHOD VALUE
- * param pname: what it do
- * returns: what it returns
- */
\ No newline at end of file
+ * Displays a greeting for the player to read.
+ */
+function sayHello() {
+    game.showLongText("Welcome to our new Adventure! You will have a good time here!", DialogLayout.Bottom);
+}
+
+sayHello();
+```
+
+## Student Task #1: Writing a Comment
+
+1. Review the code below
+2. Fill in the comment for the ``||functions:helloWorld||`` function
+3. Fill in the comment for the ``||functions:goodbye||`` function
+4. **Challenge:** should a description of ``||variables:word||`` show up in the comment for ``||functions:goodbye||``?
+
+```typescript
+/**
+ * 
+ */
+function helloWorld() {
+    game.splash("Hello World!");
```

Comment:
Saying "Hello World" doesn't really serve a real life purpose so it might be difficult for students come up with a comment description for this.

Reply:
Trying to keep it almost identical to ``sayHello`` so the first comment they right is trivially easy / basically copy paste, maybe splashing ``My name is bob`` in function ``introduction`` instead?